tests/trx_generator/trx_provider.cpp

```diff
@@ -50,28 +53,64 @@ namespace eosio::testing {
       _p2p_socket.send(boost::asio::buffer(*msg));
    }
 
-   p2p_trx_provider::p2p_trx_provider(const provider_base_config& provider_config) :
-      _peer_connection(provider_config) {
+   void http_connection::connect() {}
 
+   void http_connection::disconnect() {
+      int max    = 30;
+      int waited = 0;
+      while (_sent != _acknowledged && waited < max) {
+         ilog("http_connection::disconnect waiting on ack - sent ${s} | acked ${a} | waited ${w}",
+              ("s", _sent)("a", _acknowledged)("w", waited));
+         sleep(1);
+         ++waited;
+      }
    }
 
-   void p2p_trx_provider::setup() {
-      _peer_connection.connect();
+   void http_connection::send_transaction(const chain::packed_transaction& trx) {
+      const std::string target       = "/v1/chain/send_transaction2"s;
+      const int         http_version = 11;
+      const std::string content_type = "application/json"s;
+
+      bool        retry                = false;
+      bool        tx_rtn_failure_trace = true;
+      auto        to_send              = fc::mutable_variant_object()("return_failure_trace",
+                                                  tx_rtn_failure_trace)("retry_trx", retry)("transaction", trx);
```

Comment:
Seems like an odd line break location to me.

Reply:
Addressed: [f342f5e](https://github.com/AntelopeIO/leap/pull/1138/commits/f342f5e3cace6bd7db29e548640fc9a23fe53d43)